modules/rostests/kmtests/kmtest/service.c

```diff
@@ -100,6 +102,103 @@ KmtCreateService(
                              ServiceHandle);
 }
 
+/**
+ * @name KmtGetServiceStateAsString
+ *
+ * @param ServiceState
+ *        Service state as a number
+ *
+ * @return Service state as a string
+ */
+static
+PCSTR
+KmtGetServiceStateAsString(
+    IN DWORD ServiceState)
+{
+    switch(ServiceState)
+    {
+        case SERVICE_STOPPED:
+            return "STOPPED";
+        case SERVICE_START_PENDING:
+            return "START_PENDING";
+        case SERVICE_STOP_PENDING:
+            return "STOP_PENDING";
+        case SERVICE_RUNNING:
+            return "RUNNING";
+        case SERVICE_CONTINUE_PENDING:
+            return "CONTINUE_PENDING";
+        case SERVICE_PAUSE_PENDING:
+            return "PAUSE_PENDING";
+        case SERVICE_PAUSED:
+            return "PAUSED";
+        default:
+            return "(Unknown)";
+    }
+}
+
+/**
+ * @name KmtCheckServiceState
+ *
+ * @param ServiceName
+ *        Name of the service to check,
+ *        or NULL
+ * @param ServiceHandle
+ *        Handle to the service
+ * @param ExpectedServiceState
+ *        State which the service should be in
+ *
+ * @return Win32 error code
+ */
+static
+DWORD
+KmtCheckServiceState(
+    IN PCWSTR ServiceName OPTIONAL,
+    IN SC_HANDLE ServiceHandle,
+    IN DWORD ExpectedServiceState)
+{
+    DWORD Error = ERROR_SUCCESS;
+    SERVICE_STATUS ServiceStatus;
+    DWORD StartTime = GetTickCount();
+    DWORD Timeout = 30000; // 30 second timeout
```

Comment:
Comment adds no information. Unless you consider it non-obvious that these are milliseconds, in which case `#define SECOND 1000` and let the code speak for itself.

Reply:
Replaced with in-between solution: `30 * 1000`.

NB:
30-seconds feels pretty long to wait for, when it happens (for broken tests).
Tell me whether you want/agree to reduce it to 10 for example.

Reply:
Good point. I agree 10 seconds seems more than enough.

Reply:
Done. With `Sleep(1 * 1000);` too.